Dockerfile.cpu

```diff
@@ -0,0 +1,62 @@
+FROM ubuntu:18.04 as build-stage
+
+WORKDIR /usr/src/app
+
+ENV LANG="C.UTF-8" LC_ALL="C.UTF-8" PATH="/opt/venv/bin:$PATH" PIP_NO_CACHE_DIR="false" CFLAGS="-mavx2 -mf16c" CXXFLAGS="-mavx2 -mf16c"
+
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+    python3 python3-pip python3-venv python3-dev wget make g++ libblas-dev liblapack-dev patch \
+    build-essential zlib1g-dev libsdl2-dev libjpeg-dev nasm tar libbz2-dev libgtk2.0-dev cmake \
+    git libfluidsynth-dev libgme-dev libopenal-dev timidity libwildmidi-dev unzip libboost-all-dev \
+    libcanberra-gtk-module libopenmpi-dev libosmesa6-dev libgl1-mesa-glx libglfw3 && \
+    rm -rf /var/lib/apt/lists/*
+
+RUN export DISPLAY=:0
+
+COPY requirements.txt .
+
+RUN python3 -m venv /opt/venv && \
+    python3 -m pip install pip==19.3.1 pip-tools==4.2.0 wheel==0.33.6 setuptools==44.0.0
+
+RUN python3 -m piptools sync
+
+COPY . .
+
+FROM ubuntu:18.04 as dev-stage
+
+WORKDIR /usr/src/app
+
+ENV LANG="C.UTF-8" LC_ALL="C.UTF-8" PATH="/opt/venv/bin:$PATH" PIP_NO_CACHE_DIR="false"
+
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+    python3 python3-pip python3-venv python3-dev wget make g++ libblas-dev liblapack-dev patch \
+    build-essential zlib1g-dev libsdl2-dev libjpeg-dev nasm tar libbz2-dev libgtk2.0-dev cmake \
+    git libfluidsynth-dev libgme-dev libopenal-dev timidity libwildmidi-dev unzip libboost-all-dev \
+    libcanberra-gtk-module libopenmpi-dev libosmesa6-dev libgl1-mesa-glx libglfw3 && \
+    rm -rf /var/lib/apt/lists/*
```

Comment:
If you do compile from source above, then here you don't need the -dev packages anymore.

For package p
- libp-dev contains headers, impl., sources, etc. for compilation and linking
- libp contains runtime libs for dynamic loading

Use the dev packages for your compilation stage and the non-dev packages for your final stage.

Reply:
Ah got it.